channeldb/forwarding_log.go

```diff
@@ -287,10 +302,42 @@ func (f *ForwardingLog) Query(q ForwardingEventQuery) (ForwardingLogTimeSlice, e
 					return err
 				}
 
-				event.Timestamp = currentTime
-				resp.ForwardingEvents = append(resp.ForwardingEvents, event)
-
-				recordOffset++
+				// Check if the incoming channel ID matches the
+				// filter criteria.
+				// Either no filtering is applied (IsEmpty), or
+				// the ID is explicitly included.
+				incomingMatch := q.IncomingChanIDs.IsEmpty() ||
+					q.IncomingChanIDs.Contains(
+						event.IncomingChanID.ToUint64(),
+					)
+
+				// Check if the outgoing channel ID matches the
+				// filter criteria.
+				// Either no filtering is applied (IsEmpty), or
+				// the ID is explicitly included.
+				outgoingMatch := q.OutgoingChanIDs.IsEmpty() ||
+					q.OutgoingChanIDs.Contains(
+						event.OutgoingChanID.ToUint64(),
+					)
+
+				// If both conditions are met, then we'll add
+				// the event to our return payload.
+				if incomingMatch && outgoingMatch {
```

Comment:
Could invert this condition and do a `continue`. Then the rest of the code could continue without an additional level of indentation.